sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionPump.java

```diff
@@ -61,14 +66,23 @@ void setLastEnqueuedEventProperties(LastEnqueuedEventProperties lastEnqueuedEven
      */
     @Override
     public void close() {
+        // When client receive complete signal will call this method again in clean up method on scheduler thread,
+        // the first dispose will cause a timeout exception. Add a check avoid close twice.
+        if (isDisposed.getAndSet(true)) {
+            return;
+        }
         try {
             client.close();
         } catch (Exception error) {
             LOGGER.atInfo()
                 .addKeyValue(PARTITION_ID_KEY, partitionId)
                 .log("Exception occurred disposing of consumer client.", error);
         } finally {
-            scheduler.dispose();
+            try {
+                ReactorShim.disposeGracefully(scheduler).block(Duration.ofSeconds(1));
+            } catch (IllegalStateException e) {
+                scheduler.dispose();
```

Comment:
Can we add a small comment in the code about this - i.e.

"If the scheduler cannot be disposed of gracefully within the grace period, then an IllegalStateException is thrown, on which we do eager disposal."

Reply:
Sure!